File: code/graphs/bfs.py
from collections import deque

def bfs(g, s):
    q = deque([s])
    seen = [False] * g.V
    seen[s] = True
    while q:
        u = q.popleft()                 # visit u
        for v in g.adj[u]:
            if not seen[v]:
                seen[v] = True          # mark on ENQUEUE
                q.append(v)

File: code/graphs/bfs.rs
use std::collections::VecDeque;

fn bfs(g: &Graph, s: usize) {
    let mut q = VecDeque::from([s]);
    let mut seen = vec![false; g.adj.len()];
    seen[s] = true;
    while let Some(u) = q.pop_front() {  // visit u
        for &v in &g.adj[u] {
            if !seen[v] {
                seen[v] = true;          // mark on ENQUEUE
                q.push_back(v);
            }
        }
    }
}
